recipes/ios-merging-test-results-and-deploying-to-the-test-reports-add-on.md

```diff
@@ -0,0 +1,153 @@
+
+## Instructions
+
+1. Visit the [Create New App page](https://app.bitrise.io/apps/add) to create a new App.
+1. When prompted to select a git repository, choose **Other/Manual** and paste the sample project repository URL (`https://github.com/bitrise-io/sample-swift-project-with-parallel-ui-test`) in the **Git repository (clone) URL** field.
+1. Confirm that this is a public repository in the resulting pop-up.
+1. Select the `master` branch to scan.
+1. Wait for the project scanner to complete.
+1. Select any of the offered Distribution methods (for example development, it does not really matter as now we are focusing on testing).
+1. Confirm the offered stack, skip choosing the app icon and the webhook registration and kick off the first build.
+1. Open the new Bitrise project’s Workflow Editor. 
+1. Go to the **bitrise.yml** tab and replace the existing `bitrise.yml` with the contents of the example bitrise.yml below.
+1. Click the **Start/Schedule a Build** button, and select the `run_tests_groups` option in the **Workflow, Pipeline** dropdown menu at the bottom of the popup.
+1. Open the Pipeline’s build page.
+1. Select the `deploy_test_results` build.
+1. Click on **Details & Add-ons** on the build details page and select the Test Reports add-on to view the merged test reports.
+
+## bitrise.yml
+
+```yaml
+---
+format_version: '11'
+default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
+project_type: ios
+
+app:
+  envs:
+  - BITRISE_PROJECT_PATH: BullsEye.xcworkspace
+  - BITRISE_SCHEME: BullsEye
+
+meta:
+  bitrise.io:
+    stack: osx-xcode-13.2.x
+
+pipelines:
+  run_tests_groups:
+    stages:
+    - build_tests: {}
+    - run_tests_groups: {}
+    - deploy_test_results: {}
+
+stages:
+  build_tests:
+    workflows:
+    - build_tests: {}
+
+  run_tests_groups:
+    workflows:
+    - run_ui_tests: {}
+    - run_unit_tests: {}
+    
+  deploy_test_results:
+    workflows:
+    - deploy_test_results: {}
+
+workflows:
+  build_tests:
+    steps:
+    - git-clone: {}
+    - xcode-build-for-test:
+        inputs:
+        - destination: generic/platform=iOS Simulator
+    - deploy-to-bitrise-io: {}
+
+  run_ui_tests:
+    before_run:
+    - _pull_test_bundle
+    steps:
+    - xcode-test-without-building:
+        inputs:
+        - xctestrun: "$BITRISE_TEST_BUNDLE_PATH/BullsEye_UITests_iphonesimulator15.2-arm64-x86_64.xctestrun"
+        - destination: platform=iOS Simulator,name=iPhone 12 Pro Max
+    - deploy-to-bitrise-io: {}
+
+  run_unit_tests:
+    before_run:
+    - _pull_test_bundle
+    steps:
+    - xcode-test-without-building:
+        inputs:
+        - xctestrun: "$BITRISE_TEST_BUNDLE_PATH/BullsEye_UnitTests_iphonesimulator15.2-arm64-x86_64.xctestrun"
+        - destination: platform=iOS Simulator,name=iPhone 12 Pro Max
+    - deploy-to-bitrise-io: {}
+        
+  deploy_test_results:
+    steps:
+    - git::https://github.com/bitrise-steplib/bitrise-step-artifact-pull.git@main:
```

Comment:
I've released the artifact pull step to the steplib, so we can reference this by ID now

```suggestion
    - artifact-pull:
```